### ISSUE TITLE
REPL: tab completion: results are unordered when tabbing immediately after `.`

### ISSUE THREAD

**Issue by SethTisue** [scala/bug#12280]
if I type at least one character, completions are in alphabetical order, as expected:

```
scala 2.13.4> "foo".a<TAB>
addString(                 andThen(                   appendedAll(               applyOrElse(
aggregate(  (deprecated)   appended(                  apply(                     asInstanceOf (universal)
```

but if I tab immediately after `.`, they aren't:

```
scala 2.13.4> "foo".<TAB>
toCharArray()                       empty                               formatted(            (universal)               takeWhile(                          prefixLength(        (deprecated)   takeRight(                                      unapply(                            view                                filterNot(                                      toByte                              inits                               unzip(                                          andThen(                            toUpperCase()                       notify()              (universal)               patch(                              lazyZip(                            partitionMap(                                   iterator                            codePoints()                        toLongOption                                    prependedAll(                       last                                map(                                            toDouble                            unzip3(                             ++(                                             trim()                              count(                              :\(                  (deprecated)               mkString                            eq(                   (universal)   product(                                        sortBy(                             reverseIterator                     max(                                            equ
...
```

what the hell?

**Comment by SethTisue** [comment]
I tried adding a little test to `CompletionTest.scala`, and it seems the results are coming back in order, internally: `List(!=, ##, *, +, +, ++, ++, ++, ++, ++:, ++:, +:, +:, ->, /:, :+, :+, :++, :++, :\, <, <=, ==, >, >=, addString, addString, addString, aggregate, andThen, andThen, appended, appended, appendedAll, appendedAll, apply, applyOrElse, asInstanceOf, canEqual, capitalize, charAt, charStepper, chars, codePointAt, codePointBefore, codePointCount, codePointStepper, codePoints, collect, ...`

so is actually this some weird JLine issue? maybe it will go away if we can make the grouping UI work, as part of https://github.com/scala/bug/issues/12272 ?

**Comment by jxnu-liguobin** [comment]
It seams look good in 2.13.5.

**Comment by griggt** [comment]
This reminds me of https://github.com/lampepfl/dotty/issues/11488 which was fixed by upgrading to JLine 3.19.0.
